docs/en/tutorials/hook.md

```diff
@@ -1,3 +1,254 @@
 # Hook
 
-Coming soon. Please refer to [chinese documentation](https://mmengine.readthedocs.io/zh_CN/latest/tutorials/hook.html).
+Hook programming is a programming pattern in which a mount point is set in one or more locations of a program, and when the program runs to a mount point, all methods registered to the it at runtime are automatically called. Hook programming can increase the flexibility and extensibility of the program, and the user can register custom methods to the mount point to be called without modifying the code in the program.
```

Comment:
```suggestion
Hook programming is a programming pattern in which a mount point is set in one or more locations of a program. When the program runs to a mount point, all methods registered to it at runtime are automatically called. Hook programming can increase the flexibility and extensibility of the program, since users can register custom methods to the mount point to be called without modifying the code in the program.
```